pallets/parachain-staking/src/lib.rs

```diff
@@ -1085,6 +1266,27 @@ pub mod pallet {
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
+		#[pallet::weight(1_000_000_000)]
+		/// Temporary root function to return nominations
+		/// - charges uniform 0.2% of block weight as fee per call
+		pub fn hotfix_unreserve_nomination(
+			origin: OriginFor<T>,
+			stakers: Vec<(T::AccountId, BalanceOf<T>)>,
+		) -> DispatchResultWithPostInfo {
+			frame_system::ensure_root(origin)?;
+			let mut sum_unreserved: BalanceOf<T> = 0u32.into();
```

Comment:
Question, dont we added AccountsDueUnreservedBalance storage item to track this due unreserve balance? Is efficiency the reason why we are providing the due_unreserve balance as a parameter (I mean we could look it up in the aformentioned storage item but that would imply a DBread). And if so, is the storage item necessary?

Reply:
> Question, dont we added AccountsDueUnreservedBalance storage item to track this due unreserve balance? Is efficiency the reason why we are providing the due_unreserve balance as a parameter

It doesn't have all the due unreserved balances and only stores a subset of all nominations that have been bumped out due to this bug. If a nominator left, then the map will set `0` as the value for the nomination.

The storage item's only purpose is to cache all nominations that were removed from `top_nominators` and `bottom_nominators`. Then I get it while iterating over each nominator's nominations to see if the nomination must be removed and the amount due (map's value) can be updated.

In conclusion, if the value for `AccountsDueUnreservedBalance` is `0` then the nominator some due unreserved balance. If it is set to an amount, then it didn't leave the set of nominators so the set amount represents what they're actually due.

Reply:
@4meta5 keep in mind that some nominator have been removed from the list  but might have been re-introduced after and still have the wrong unreserved. (However when I tested yesterday, the hotfix_unreserve_... worked)

Reply:
But why cache into a storage item and not into a temporary btree map? Is it the case that our fix script will need to read this storage item to be able to introduce it as a parameter in `hotfix_unreserve_nomination`?

Reply:
Good point, the fix script isn't using this storage item at all. I'm going to do exactly what you just said now. It's kind of messy based on how I've written this because we need to pass the BTreeMap around.